app/views/incomeSources/manage/ManageIncomeSources.scala.html

```diff
@@ -0,0 +1,172 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import _root_.implicits.ImplicitDateFormatterImpl
+@import views.html.layouts.unifiedLayout
+@import views.html.components._
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import models.incomeSourceDetails.viewmodels.ViewIncomeSourcesViewModel
+
+
+@this(
+        mainTemplate: unifiedLayout,
+        implicitDateFormatter: ImplicitDateFormatterImpl,
+        h1: h1,
+        h2: h2,
+        p: p,
+        link: link,
+        hr: hr,
+        govukTable: GovukTable
+)
+
+@(
+        sources: ViewIncomeSourcesViewModel,
+        isAgent: Boolean,
+        backUrl: String
+)(implicit messages: Messages, user: auth.MtdItUser[_])
+
+@import implicitDateFormatter.longDate
+
+@getMessage(key: String, args: String*) = @{
+    messages(s"view-income-sources.$key", args: _*)
+}
+
+@viewSelfEmploymentDetailsURL = @{
+    if(isAgent){
+        controllers.incomeSources.manage.routes.ManageSelfEmploymentSourceController.showAgent().url
+    } else {
+        controllers.incomeSources.manage.routes.ManageSelfEmploymentSourceController.show().url
+    }
+}
+
+@ukProperty = @{ sources.viewUkProperty }
+@foreignProperty = @{ sources.viewForeignProperty }
+@ceasedBusinesses = @{ sources.viewCeasedBusinesses }
+@soleTraderBusinesses = @{ sources.viewSoleTraderBusinesses }
+
+
+@mainTemplate(pageTitle = getMessage("heading"),
+    backUrl = Some(backUrl),
+    btaNavPartial = user.btaNavPartial) {
+
+    @h1(msg = getMessage("heading"),
+        id = Some("new-income-sources-h1"))
+
+    @if(soleTraderBusinesses.nonEmpty) {
+        @h2(msg = getMessage("self-employed-h2"),
+            optId = Some("self-employed-h1"))
+
+        @govukTable(Table(
+            head = Some(List(
+                HeadCell(
+                    content = Text(getMessage("table-head.business-name")),
+                    attributes = Map("id" -> "table-head-business-name")),
+                HeadCell(
+                    content = Text(getMessage("table-head.date-started")),
+                    attributes = Map("id" -> "table-head-date-started")),
+                HeadCell(
+                    content = Text("") //TODO see on extending line w/o blank cell
+                )
+            )),
+            rows = soleTraderBusinesses.zipWithIndex.map { case (business, index) => Seq(
+                TableRow(
+                    content = Text(business.tradingName),
+                    attributes = Map("id" -> s"table-row-trading-name-$index")),
+                TableRow(
+                    content = Text(business.tradingStartDate.map(_.toLongDate).getOrElse("Unknown")),
```

Comment:
as discussed, we need to support both language, even you might not received translation for Welsh, this must be handled according to the language selected by the user

Reply:
Done